awx/main/dispatch/worker/callback.py

```diff
@@ -189,6 +195,11 @@ def flush(self, force=False):
                 self.subsystem_metrics.observe('callback_receiver_batch_events_insert_db', metrics_bulk_events_saved)
                 self.subsystem_metrics.inc('callback_receiver_events_in_memory', -(metrics_bulk_events_saved + metrics_singular_events_saved))
                 self.subsystem_metrics.inc('callback_receiver_events_broadcast', metrics_events_broadcast)
+                self.subsystem_metrics.set(
+                    'callback_receiver_event_processing_avg_seconds',
+                    metrics_total_job_event_processing_seconds.total_seconds()
+                    / (metrics_bulk_events_saved + metrics_singular_events_saved - metrics_events_missing_created),
+                )
             if self.subsystem_metrics.should_pipe_execute() is True:
```

Comment:
@rebeccahhh @AlanCoding 
...so rebeccah sent me examples of how this looks now that we dropped the `else` condition and we never zero out the metric.
![image](https://user-images.githubusercontent.com/29239279/171852644-1cbe4755-6853-4836-a5c1-26d273736ae8.png)
![image (1)](https://user-images.githubusercontent.com/29239279/171852704-bf6208a1-175f-48e4-976d-3a49dabe7b8b.png)





TBH I think it is much worse, and would make it difficult/harder to use in a sane way because it does exactly what alan suggested which is leave it pegged up high. Seems like there was some concern that hypothetically having the else condition would cause this situation where the the metric would oscillate between zero and some number because there would be one idle worker while others were posting a value. We never actually have seen that happen though. So to me, its the lesser of the evils because with leaving the metric up high you impede usefulness of any computation of avg, mean, and stdev. And if we later do what @AlanCoding wants to to which is have the parent process be the one that really pushes the metrics, this won't be a concern at all.

Reply:
to further add onto this, this will actually never zero out (I tested it overnight, lag hovered just above 1 second for the whole night, having taken the average from many hours earlier and running with it and it's lack of change) which I find further problematic in the short term. I added the else statement back on because as @kdelee mentioned, this won't be an issue at all in a future PR. I left a comment so we know to adjust that section of the code when the time comes.

Reply:
Here's a visual example where you can see from yesterday afternoon until about 8am it never changes. It's just not useful information at that point in my opinion. Since we're taking an average I would rather have some zeros included when appropriate vs. this situation.
![image](https://user-images.githubusercontent.com/11764497/171952750-b8557231-749e-4e33-a916-c6583c984ead.png)

Reply:
Compromising _performance itself_ for the visuals of performance metrics is like throwing the baby out with the bathwater. The current `else:` strikes me as a categorically bad decision.

If you wish to set it back down to 0, then you should not continue to update it to that value on every flush. This wouldn't be a problem if the subsystem metrics utils, themselves, compared new values to old values and avoided unnecessary system calls when nothing changed. That doesn't appear to be what it does.

https://github.com/ansible/awx/blob/d9f26a411ecd893725372b93b9698b017a40b739/awx/main/analytics/subsystem_metrics.py#L195-L197

If you still wish to set it down to 0 in this event, you should track a variable on `self` to remember whether or not the last value is already 0, and don't keep updating it if it is.